### PR TITLE
Ls fixing customize user location example

### PR DESCRIPTION
**Pull request by langsmith**
Fixes https://github.com/mapbox/mapbox-android-demo/issues/211

ezgif com-resize

### CODE REVIEW

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java [hunk 1]
```diff
@@ -24,6 +29,12 @@
   private PermissionsManager permissionsManager;
   private MapView mapView;
   private MapboxMap map;
+  private LocationEngine locationEngine;
+  private LocationEngineListener locationEngineListener;
+
+
```
Comment (cammace) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java]: nit

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java [hunk 2]
```diff
@@ -52,14 +62,15 @@ public void onMapReady(MapboxMap mapboxMap) {
...
-          enableGps();
+          enableLocation();
         }
 
         // Customize the user location icon using the getMyLocationViewSettings object.
         map.getMyLocationViewSettings().setPadding(0, 500, 0, 0);
         map.getMyLocationViewSettings().setForegroundTintColor(Color.parseColor("#56B881"));
         map.getMyLocationViewSettings().setAccuracyTintColor(Color.parseColor("#FBB03B"));
 
+
```
Comment (cammace) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java]: nit

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java [hunk 3]
```diff
@@ -113,22 +124,63 @@ private void enableGps() {
   }
 
   @Override
-  public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-    permissionsManager.onRequestPermissionsResult(requestCode, permissions, grantResults);
+  public void onRequestPermissionsResult(
+    int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+    if (requestCode == PERMISSIONS_LOCATION) {
```
Comment (cammace) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java]: Any reason for switching from using the PermissionsManager here?

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java [hunk 4]
```diff
@@ -113,22 +124,63 @@ private void enableGps() {
...
+          // listener is registered again and will adjust the camera once again.
+          map.moveCamera(CameraUpdateFactory.newLatLngZoom(new LatLng(location), 16));
+          locationEngine.removeLocationEngineListener(this);
+        }
+      }
+    };
+
+    locationEngine.addLocationEngineListener(locationEngineListener);
+
+
```
Comment (cammace) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java]: nit

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java [hunk 5]
```diff
@@ -113,22 +124,63 @@ private void enableGps() {
...
+    //Enable or disable the location layer on the map
+    map.setMyLocationEnabled(true);
+  }
+
+
```
Comment (cammace) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java]: nit

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java (hunk 3 above)
Comment (langsmith) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java]: Forgot to switch back to using the PermissionsManager amongst my experimenting. Latest commit uses it.

**Comment by langsmith** [comment]
Requested changes have been fixed @cammace

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java [hunk 6]
```diff
@@ -24,6 +29,10 @@
   private PermissionsManager permissionsManager;
   private MapView mapView;
   private MapboxMap map;
+  private LocationEngine locationEngine;
+  private LocationEngineListener locationEngineListener;
+  private static final int PERMISSIONS_LOCATION = 0;
```
Comment (cammace) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java]: Doesn't look like this constants being used anymore

Comment (langsmith) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/location/CustomizeUserLocationActivity.java]: Fixed

**Pull request merged**